### PR TITLE
Initial rewrite of zignatures

### PR DESCRIPTION
**Pull request by jroimartin**
Includes:
* Add exact and anal zignatures (zae, zaa)
* Delete zignatures (z-)
* Manage zignspaces (zs)
* Search zignatures (z/)
* Check zignature at address (zc)
* FLIRT signatures (zf)

Missing:
* zo
* zfz
* zam
* Honor search.in and search.align

### CODE REVIEW

File: libr/anal/sign.c [hunk 1]
```diff
@@ -1,235 +1,357 @@
...
+static bool deserialize(RSignItem *it, const char *k, const char *v) {
+	char *k2 = NULL, *v2 = NULL, *ptr = NULL, *token = NULL;
+	int i = 0;
+	bool retval = true;
+
+	k2 = r_str_new (k);
+	v2 = r_str_new (v);
+
+	// Deserialize key: zign,<space>,<name>
+	for (ptr = (char*) k2, i = 0; ; ptr = NULL, i++) {
```
Comment (ret2libc) [inline libr/anal/sign.c]: no need for cast

File: libr/anal/sign.c [hunk 2]
```diff
@@ -1,235 +1,357 @@
...
-		}
-		break;
-	case R_SIGN_HEAD: // function prefix (push ebp..)
-	case R_SIGN_BYTE: // function mask
-	case R_SIGN_BODY: // function body
-		if (!(data = r_anal_strmask (anal, arg))) {
-			r_sign_item_free (si);
-			si = NULL;
+	// Deserialize val: <type>,size,bytes,mask
+	for (ptr = (char*) v2, i = 0; ; ptr = NULL, i++) {
```
Comment (ret2libc) [inline libr/anal/sign.c]: no need for cast

File: libr/anal/sign.c [hunk 3]
```diff
@@ -1,235 +1,357 @@
...
+	struct ctxDeleteCB *ctx = (struct ctxDeleteCB *)user;
+
+	if (r_str_cmp (k, ctx->buf, strlen (ctx->buf))) {
+		sdb_remove (ctx->anal->sdb_zigns, k, 0);
+	}
+
+	return 1;
+}
+
+R_API bool r_sign_delete(RAnal *a, const char *name) {
```
Comment (ret2libc) [inline libr/anal/sign.c]: What about removing `const`? It looks like it's not guaranteed to be const.

Comment (jroimartin) [inline libr/anal/sign.c]: Actually it is guaranteed. The string shouldn't be changed from this function because the name is only used to query sdb.

Comment (ret2libc) [inline libr/anal/sign.c]: Mmm at https://github.com/radare/radare2/pull/6997/files#diff-14db28e2b566436a8da7e19ec6c2ea75R174 you are removing the cast. Could it be changed thorough that pointer?

Comment (jroimartin) [inline libr/anal/sign.c]: It's not changed, however it's a bit tricky I must admit. I'll check it. Thx!

**Pull request merged**